src/UI/Art/SpriteArtist/SpriteMiniMapArtist.java

```diff
@@ -55,7 +41,7 @@ private static BufferedImage generateMapImage(MapInfo mapInfo)
     {
       for(int x = 0; x < image.getWidth(); ++x)
       {
-        g.setColor( terrainColors.get(mapInfo.terrain[x][y].index) );
+        g.setColor( mapInfo.terrain[x][y].getMainColor() );
```

Comment:
Perhaps I should have looked more closely at this before my initial change... oh, well. Good catch.